src/ios-deploy/ios-deploy.m

```diff
@@ -91,6 +91,9 @@
 mach_error_t AMDeviceLookupApplications(AMDeviceRef device, CFDictionaryRef options, CFDictionaryRef *result);
 int AMDeviceGetInterfaceType(AMDeviceRef device);
 AMDeviceRef AMDeviceCopyPairedCompanion(AMDeviceRef device);
+#if defined(XCODE_14_OR_NEWER_AVAILABLE)
```

Comment:
What happens if you compile this on Xcode14 and then run it against Xcode12? Wouldn't AMDeviceCopyDeveloperModeStatus not exist?

Reply:
It would still work because there is a single MobileDevice.framework that gets updated along with each Xcode release.   As long as the user installed Xcode 14 at one point and installed the developer tools, this feature will work.

Reply:
What about centrally compiled versions? e.g. I believe Flutter and Cordova compile the ios-deploy binary on their server and then distribute that binary.

Reply:
I'm not sure, I don't have a machine that hasn't had Xcode 14 installed but I think that would result in a crash when the program attempts to use the invalid symbol. I don't know of a good way around that so for that reason this feature is disabled by default and requires building with "-DXCODE_14_OR_NEWER_AVAILABLE=1" to enable. Then any tools that invoke ios-deploy would need to perform their own checks that Xcode 14 is available before attempting to use this feature.

Reply:
For compiling it would be best to switch to a tbd file, so we don't have to gate on the version of Xcode used to compile, but requiring Xcode14 to compile is also an option.

For run-time checking I'd just put these calls not in the main path. If it fails we should try to tell them they need Xcode14, but it's fine if it fails.

Reply:
I'm not sure what "switching to a tbd file" means. ios-deploy won't compile if Xcode 14 has not been installed to the system because the MobileDevice framework that is dynamically linked won't have the symbol defined (the reason why I rolled back the previous commit). A tbd would fail the same way. 

For runtime, I guess we could dlopen and dlsym to check if the symbols exists before we attempt to call it but simply noting the requirement in the usage + README.md seems sufficient to me

Reply:
From our discussion offline, it seemed we had a misunderstanding over when the define would be defined, and by extension when the feature would be enabled. I've renamed `XCODE_14_OR_NEWER_AVAILABLE` to `IOS_DEPLOY_FEATURE_DEVELOPER_MODE`  so it is clear that this define must be manually provided in order to enable the developer mode option. This feature is opt-in because it requires Xcode 14 on the machine that builds ios-deploy AND on the machine that runs it.